soh/src/overlays/actors/ovl_En_Wallmas/z_en_wallmas.c

```diff
@@ -304,13 +304,17 @@ void EnWallmas_WaitToDrop(EnWallmas* this, PlayState* play) {
     if (this->timer != 0) {
         this->timer--;
     }
-
-    if ((player->stateFlags1 & 0x100000) || (player->stateFlags1 & 0x8000000) || !(player->actor.bgCheckFlags & 1) ||
-        ((this->actor.params == 1) && (320.0f < Math_Vec3f_DistXZ(&this->actor.home.pos, playerPos)))) {
+    if (this->actor.params == WMT_SHADOWTAG) {
+        if ((player->stateFlags1 & 0x100000) || (player->stateFlags1 & 0x8000000) || !(player->actor.bgCheckFlags & 1)) {
+            Audio_StopSfxById(NA_SE_EN_FALL_AIM);
+            this->timer = 0x82;
+        }
+    } else if ((player->stateFlags1 & 0x100000) || (player->stateFlags1 & 0x8000000) || !(player->actor.bgCheckFlags & 1) ||
+            ((this->actor.params == 1) && (320.0f < Math_Vec3f_DistXZ(&this->actor.home.pos, playerPos)))) {
         Audio_StopSfxById(NA_SE_EN_FALL_AIM);
         this->timer = 0x82;
-    }
-
+    } 
+    
```

Comment:
One part of me thinks there should be a way of not having duplicated code in here, other part of me hates how that if statement would look like. Open for other's opinions.